mailpoet/mailpoet
# Update @wordpress and React packages [MAILPOET-5102]

## Description
This PR updates @wordpres/* and React packages. React update is included because some of the wordpress packages [have already updated to React 18](https://github.com/WordPress/gutenberg/pull/45235) and use the new functionality.

Note: After the update, there is the dev console for dev builds.
```
Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot
```
This is expected for now and will be addressed in a follow-up ticket. Switching to the new render API will turn on the automatic batching feature, and it may cause some issues.


## Code review notes

_N/A_

## QA notes

This changes a lot of JS libraries we build on, so potentially, this could cause issues in the whole plugin.
Five parts of the plugin are influenced the most because they heavily use these packages:

1. Automation - some visual changes are expected
2. Form Editor - some visual changes are expected
3. Plugin settings
4. Dynamic segments editor (the form for creating segments)
5. Settings panel of the coupon block - some small visual changes are expected

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/692 - related changes in the premium plugin
https://github.com/mailpoet/wordpress-images/pull/43 - update of the image for CI builds

## Linked tickets

[MAILPOET-5102]

## After-merge notes

This also updates the node version in the dev environment so ping devs to update the local node and rebuild the WordPress image in their dev env.


[MAILPOET-5102]: https://mailpoet.atlassian.net/browse/MAILPOET-5102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
